app/scripts/views/sign_up.js

```diff
@@ -218,6 +222,15 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
       return this._isMailcheckEnabledValue;
     },
 
+    passwordCheck: function () {
+      var self = this;
+      var password = this.getElementValue('.password');
+      this.passwordcheck(password, function (msg) {
+        self.logScreenEvent(msg);
```

Comment:
Any `msg` returned from `passwordCheck` that indicates an error should be converted to an [AuthErrors](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/lib/auth-errors.js) based error, with a [client-side originating errno](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/lib/auth-errors.js) (1000+). Use [message: UNEXPECTED_ERROR_MESSAGE](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/lib/auth-errors.js#L215) as the message until we decide what text we are going to show in the tooltip.

For errors, instead of using `self.logScreenEvent`, use [self.logError](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/views/base.js#L476).